wandb/sdk/launch/launch.py

```diff
@@ -0,0 +1,199 @@
+    :param api: An instance of a wandb Api from wandb.apis.internal.
+    :param entry_point: Entry point to run within the project. Defaults to using the entry point used
+                        in the original run for wandb URIs, or main.py for git repository URIs.
+    :param version: For Git-based projects, either a commit hash or a branch name.
+    :param parameters: Parameters (dictionary) for the entry point command. Defaults to using the
+                       the parameters used to run the original run.
+    :param docker_args: Arguments (dictionary) for the docker command.
+    :param experiment_name: Name of experiment under which to launch the run.
+    :param resource: Execution backend for the run: W&B provides built-in support for "local" backend
+    :param wandb_project: Target project to send launched run to
+    :param wandb_entity: Target entity to send launched run to
+    :param config: A dictionary which will be passed as config to the backend. The exact content
+                           which should be provided is different for each execution backend
+    :param synchronous: Whether to block while waiting for a run to complete. Defaults to True.
+                        Note that if ``synchronous`` is False and ``backend`` is "local", this
+                        method will return, but the current process will block when exiting until
+                        the local run completes. If the current process is interrupted, any
+                        asynchronous runs launched via this method will be terminated. If
+                        ``synchronous`` is True and the run fails, the current process will
+                        error out as well.
+    :return: :py:class:`wandb.launch.SubmittedRun` exposing information (e.g. run ID)
+             about the launched run.
+    .. code-block:: python
+        :caption: Example
+        import wandb
+        project_uri = "https://github.com/wandb/examples"
+        params = {"alpha": 0.5, "l1_ratio": 0.01}
+        # Run W&B project and create a reproducible docker environment
+        # on a local host
+        api = wandb.apis.internal.Api()
+        wandb.launch(project_uri, api, parameters=params)
+    .. code-block:: text
+        :caption: Output
+        ...
+        ...
+        Elasticnet model (alpha=0.500000, l1_ratio=0.010000):
+        RMSE: 0.788347345611717
+        MAE: 0.6155576449938276
+        R2: 0.19729662005412607
+        ... wandb.launch: === Run (ID '6a5109febe5e4a549461e149590d0a7c') succeeded ===
+    """
+    if docker_args is None:
+        docker_args = {}
+    if _is_wandb_local_uri(api.settings("base_url")):
+        if sys.platform == "win32":
+            docker_args["net"] = "host"
+        else:
+            docker_args["network"] = "host"
+        if sys.platform == "linux" or sys.platform == "linux2":
+            docker_args["add-host"] = "host.docker.internal:host-gateway"
```

Comment:
Instead of messing with the network and host we may just want to automatically replace "localhost" or "127.0.0.1" in "base_url" with `host.docker.internal`...

Reply:
I was under the impression that these arguments were needed by docker to use the `host.docker.internal`. Removed.